Update bug issue template to use forms

annezazu opened the pull request:
## Description
This will help address part of this open issue to move bug reports to forms: https://github.com/WordPress/gutenberg/issues/33394 I followed the documentation linked and tried to match it to the current issue template as closely as possible. I purposefully left out adding labels and customizing the title as I still believe both of those are too complex decisions to make in such a generalized way. 

Of note:

> If you want to use the same file name for your YAML issue form, you must delete the Markdown issue template when you commit the new file to your repository.

This PR doesn't include that file being removed but can be if we'd like to keep the same file name. I've never created one of these issue template forms before so might have approached this incorrectly :) Guidance appreciated. 

## Types of changes

Enhancement to project management

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->

annezazu committed:
Update bug issue template to use forms

This will help address part of this open issue to move bug reports to forms: https://github.com/WordPress/gutenberg/issues/33394

youknowriad commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -0,0 +1,149 @@
+    description: Please make sure you have confirmed it's not a theme specific problem.
+    options:
+    - label: I have tested with a default theme.
+      required: true
```
Makes me think there's too much checkboxes to create a bug report. Just want to make sure we want that or if we can leave the most important one?

For instance, there's a check about testing with the latest Gutenberg version and there's a Gutenberg version selector in the end. It's kind of contradictory.

annezazu replied:
Agreed - I was trying to match as closely as possible what was in the prior form to start but let's simply this :) I'll update and remove the latter pieces.

annezazu replied:
Thinking about this more, decided to keep the fields where people can enter versions as it's more important that we know what versions someone is using rather than relying on the checkbox since we might not get to a report for some time.

youknowriad commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -0,0 +1,149 @@
+    required: true
+- type: textarea
+  attributes:
+    label: Current Behavior
```
this one seems duplicated.

youknowriad commented:
Yeah, I think we should just go ahead and remove the old template in this PR. I don't know the new format enough, so I think it's fine to try and fix.

annezazu committed:
Remove redundant checkboxes

Removes some of the checkboxes at the top and cleans up some typos/an accidental duplicate.

annezazu committed:
Delete prior bug report template

This will remove the previous bug report template in order to use the same file name per the documentation for YAML.

annezazu commented:
Added in a commit to remove the prior form (Bug_report.md). Hopefully, this will do the trick :)

Mamaduka commented:
@annezazu I see an error when view new template - https://github.com/WordPress/gutenberg/blob/trunk/.github/ISSUE_TEMPLATE/Bug_report.yml#L7

annezazu commented:
Looks like this PR caused the main bug report template to disappear: https://github.com/WordPress/gutenberg/issues/new/choose I'm not seeing anything in the docs that might help give context. Here's the specific one for template forms: https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-issue-forms

Perhaps rather than a yml extension it needs md still? I assumed it would need to be yml for this to work.

priethor commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -0,0 +1,120 @@
+- type: textarea
+  attributes:
+    label: WordPress Information
+    description: |
+     Please list what WordPress version you are using. You can find this information in Tools → Site Health → Info → WordPress
+    required: false
+    
+- type: textarea
+  attributes:
+    label: Gutenberg Information
+    description: |
+     Please list what Gutenberg version you are using. If you aren't using Gutenberg, please note that it's not installed.
+    required: false
+    
+  - type: dropdown
+    id: browsers
+    attributes:
+      label: What browsers are you seeing the problem on?
+      multiple: true
+      options:
+        - Firefox
+        - Chrome
+        - Safari
+        - Microsoft Edge
+        - Other
+
+- type: textarea
+  attributes:
+    label: Device Information
+    description: |
+     Please list what device you are using e.g. "Desktop" or "iPhone 11".
+    required: false
+    
+    
+- type: textarea
+  attributes:
+    label: Operating System Information
+    description: |
+     Please list what operating system you are using e.g. "Windows 10" or "iOS 14" 
+    required: false
```
I feel having these many inputs can cause a bit of friction and having a single text area for all the device information would be more comfortable to use. Because of the space each text area requires, the form feels too long and uninviting and the resulting bug report ends up with many headings and sections for very short items like "Desktop" or "Windows 10".

priethor commented on .github/ISSUE_TEMPLATE/Bug_report.yml:
```diff
@@ -0,0 +1,120 @@
+     Please list what Gutenberg version you are using. If you aren't using Gutenberg, please note that it's not installed.
+    required: false
+    
+  - type: dropdown
```
I played with the dropdown idea as well and found it is more restrictive than convenient, as reports can't include:
- The browser version (e.g., is it Legacy Edge or chrome-based Edge?)
- The actual browser if it's not listed and the user selects "Other".

In the same line as my other comment, I think a single "Device Information" text area would yield better results.

priethor reviewed (commented):
By looking at bugs reported after introducing this form, it seems the device information parts are causing a bit of friction and are usually skipped. When filled, however, they result in an (in my opinion) very long report.
